### DISCUSSION TITLE
More prompt environments/tags instead of just `active`

### DISCUSSION THREAD

**Discussion by ladislasdellinger** [langfuse/langfuse#1321] [Ideas]
### Describe the feature or potential improvement

Currently, we can only promote a prompt to production. That means that we can't use a single Langfuse instance for all our environments.

There is for me two possibilities to improve this part:
- add the possibility to select for which environment a prompt is available (testing, staging, prod, etc.)
- add the possibility to create branch instead of version (like git versionning, or like what's available in https://www.prompthub.us/).

Any plan to add this ?

**Comment by marcklingen** [comment]
Hi @ladislasdellinger, we had tags (similar to docker/git) on the backlog already. Branching is also a great idea, but probably more complex. Happy to track interest for this on this post.

**Reply by ladislasdellinger** [reply]
Nice !
Do you have any idea when this will be implemented?

**Reply by marcklingen** [reply]
Can't state a timeline as I don't want to disappoint you in case something big comes up in the meantime

**Reply by ladislasdellinger** [reply]
Alright ! Thanks for the answer :)

**Reply by lmyslinski** [reply]
I'm trying to come up with a workaround in the meantime, but I see there's no API for listing prompt versions either ;/

**Reply by marcklingen** [reply]
yep, currently it was assumed that most usage is focused on the active/prod prompt. will try to ship more environments/tags soon, open for PRs on a list endpoint

**Reply by lmyslinski** [reply]
I'll be looking into that shortly

**Reply by lmyslinski** [reply]
@marcklingen I've published a draft but I need some guidance from your end regarding the API design: https://github.com/langfuse/langfuse/pull/1402#issuecomment-1992241844

**Comment by marcklingen** [comment]
Forgot to update this post, this was released back in May

https://langfuse.com/changelog/2024-05-07-prompts-api-and-deployment-labels
